src/service/persons.with.signficant.control.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Session } from "@companieshouse/node-session-handler";
+import { Request } from "express";
+import { makeApiCallWithRetry } from "./retry.handler.service";
+import { createAndLogErrorRequest, logger } from "../utils/logger";
+import { CompanyPersonsWithSignificantControlResource } from "@companieshouse/api-sdk-node/dist/services/company-psc/types";
+
+export const getCompanyPsc = async (
+  req: Request,
+  companyNumber: string,
+): Promise<CompanyPersonsWithSignificantControlResource | undefined> => {
+  const response = await makeApiCallWithRetry(
+    "companyPsc",
+    "getCompanyPsc",
+    req,
+    req.session as Session,
+    companyNumber
+  );
+
+  if (response.httpStatusCode === 404) {
+    logger.debugRequest(req, `No company PSC data found for ${companyNumber}`);
+  } else if (response.httpStatusCode >= 400) {
```

Comment:
I think you should make sure you get a 200 OK else error.

Reply:
What if you get 307 say?

Reply:
None of the other services check explicitly for 200, 200-300 are successful HTTP codes, so if the API is changed for whatever reason decides to use a different one then it'd break our web code if we explicitly check for 200. 307 is a successful HTTP code generally used for re-direction.

Reply:
Please see overseas.entities.service.ts

Reply:
Ok, we're inconsistent across our services then!
Will add that check for 200. If this API endpoint decides to change its success response to use 201 as well then we'll need changes to web side.